CONTRIBUTING.md

```diff
@@ -139,6 +139,60 @@ Then force push your branch:
 
 `git push --force origin test-branch`
 
+### Fixing sanity check failures after public API changes
+
+If your PR includes any changes to the Gradle Public API, it will cause the binary compatibility check to fail.
+The binary compatibility check runs as a part of the broader sanity check.
+The latter runs on every PR and is a prerequisite for merging.
+
+If you run the sanity check locally with the `./gradlew sanityCheck`, you can see the binary compatibility error in the output.
+It looks like the following:
+
+```
+Execution failed for task ':architecture-test:checkBinaryCompatibility'.
+> A failure occurred while executing me.champeau.gradle.japicmp.JApiCmpWorkAction
+   > Detected binary changes.
+         - current: ...
+         - baseline: ...
+     
+     See failure report at file:///<path to Gradle checkout>/subprojects/architecture-test/build/reports/binary-compatibility/report.html
+```
+
+Here are the steps to resolve the issue:
+
+1. Open the failure report mentioned in the output.\
+If you don't see the report link in the output in the IDE, make sure to select the top-level node in the structured output panel.
+The report will explain the errors in detail.
+Perhaps, you forgot to add an `@Incubating` annotation or `@since` in the javadoc.
+
+2. Accept the changes.\
+If you are sure that the changes are intentional, follow the steps described in the report.
+This includes adding the description of the changes to the `accepted-public-api-changes.json` file, and providing a reason for each change.
```

Comment:
Could you add a sample of a (good) reason?

Reply:
I am not sure if there is a good sample for a reason. Usually, the reasons are specific to each situation. Moreover, public API changes are historically done only by the team, as they stem from an evolution of some part of the build tool.

In any case, it's probably best to look at the git history of the file to get an idea what usually goes into the reason section.